metadata/src/main/java/org/apache/kafka/metadata/authorizer/StandardAuthorizerData.java

```diff
@@ -534,49 +535,19 @@ Iterable<AclBinding> acls(AclBindingFilter filter) {
     }
 
     class AclIterable implements Iterable<AclBinding> {
-        private final AclBindingFilter filter;
+        private final List<AclBinding> aclBindingList;
 
         AclIterable(AclBindingFilter filter) {
-            this.filter = filter;
+            this.aclBindingList = aclsByResource
```

Comment:
Just to clarify my understanding.

Previously, we were wrapping the aclsByResource iterator. This was intended to be thread-safe, but as you mentioned offline, there was no guard against the underlying map getting modified during iteration (since the ConcurrentSkipListMap might show updated elements to the iterator).

Instead, we are now making a copy of the matching AclBinding and returning that's lists iterator the caller. So, basically trading off memory for consistency.

Reply:
Yes, you're right. I think there is no other way we can guarantee the consistency here other than giving an Iterable that stays constant to the client accessing `Authorizer#acls`.